app/Models/MapMarker.php

```diff
@@ -141,6 +156,52 @@ public function isPolygon(): bool
         return $this->shape_id == MapMarker::SHAPE_POLY && !empty($this->custom_shape);
     }
 
+    /**
+     * Determine the type of the marker
+     * @return string
+     */
+    public function typeLabel(): string
+    {
+        if ($this->isPolygon()) {
+            return __('maps/markers.tabs.polygon');
+        } elseif ($this->isLabel()) {
+            return __('maps/markers.tabs.label');
+        } elseif ($this->isCircle()) {
+            return __('maps/markers.tabs.circle');
+        }
+        return __('maps/markers.tabs.marker');
+    }
+
+    /**
+     * Determine the icon of the marker
+     * @return string
+     */
+    public function datagridMarkerIcon(): string
+    {
+        if ($this->icon == 5 || $this->icon == 4) {
+            return '';
+        }
+
+        $icon = '<i class="fa-solid fa-map-pin"></i>';
+
+        $campaign = CampaignLocalization::getCampaign();
+        if (!empty($this->custom_icon) && $campaign->boosted()) {
+            if (Str::startsWith($this->custom_icon, '<i ')) {
+                $icon = $this->custom_icon ;
+            } elseif (Str::startsWith($this->custom_icon, ['fa-', 'ra '])) {
+                $icon = ' <i class="' . $this->custom_icon . '" aria-hidden="true"></i>';
+            } elseif (Str::startsWith($this->custom_icon, '<?xml')) {
+                $icon = 'L.Util.template(<div class="custom-icon">' . $this->resizedCustomIcon() . '</div>)';
```

Comment:
This needs to be tested and fixed ;-)
![image](https://user-images.githubusercontent.com/3263440/197285127-72d5259c-5930-4ff0-beda-17c3e08a133c.png)